src/lib/courses.ts

```diff
@@ -1,179 +1,80 @@
-              avatar_url
-            }
-            comment {
-              id
-              state
-              hide_url
-              restore_url
-              prompt
-              comment
-            }
-          }
-        }
-        lessons {
-          id
-          slug
-          title
-          description
-          path
-          icon_url
-          duration
-          thumb_url
-        }
-        instructor {
-          id
-          full_name
-          slug
-          avatar_url
-          avatar_64_url
-          bio_short
-          twitter
-        }
-        primary_tag {
-          name
-          image_url
-          slug
-        }
-      }
-    }
-  `
-  const {course} = await request(config.graphQLEndpoint, query, {slug})
-
-  return course
-}
-
-export async function loadAllCourseByPage(retryCount = 0): Promise<any> {
-  const query = /* GraphQL */ `
-    query PagedCourses($page: Int!, $per_page: Int!) {
-      courses(page: $page, per_page: $per_page) {
-        data {
-          slug
-          title
-          average_rating_out_of_5
-          watched_count
-          path
-          image_thumb_url
-          description
-          summary
-          tags {
-            name
-            label
-            image_url
-          }
-          instructor {
-            id
-            full_name
-            path
-          }
-        }
-        count
-        current_page
-        total_pages
-      }
-    }
-  `
-  try {
-    let currentPage = 1
-    let allCourses: any[] = []
-    let hasNextPage = true
-
-    while (hasNextPage) {
-      const {
-        courses: {data, count},
-      } = await request(config.graphQLEndpoint, query, {
-        page: currentPage,
-        per_page: 25,
-      })
-
-      currentPage = currentPage + 1
-      allCourses = [...allCourses, ...data]
-
-      console.debug(`\n\n~> loading courses: ${allCourses.length}/${count}\n`)
-
-      hasNextPage = allCourses.length < count
-    }
-
-    return allCourses
-  } catch (error) {
-    if (retryCount <= 4) {
-      return loadAllCourseByPage(retryCount + 1)
-    } else {
-      throw error
+const courseQuery = groq`
+*[_type == 'resource' && externalId == $courseId][0]{
+  title,
+  challengeRating,
+  description,
+  summary,
+  essentialQuestions[]->{
+    question 
+   },
+  "pairWithResources": related[]->{
+    name,
+    title,
+    byline,
+    description,
+    summary,
+    path,
+    image,
+    'instructor': collaborators[]->[role == 'instructor']{
+    	'name': person->name,
+  	}
+	},
+	"freshness": staffReviews[0]{
+    status,
+    title,
+    summary
+  },
+	"customOgImage": images[label == 'og-image'][0]{
+    url
+  },
+	'illustrator': collaborators[]->[role == 'illustrator'][0]{
+    'name': person->name,
+    'image': person->image.url
+  },
+	'instructor': collaborators[]->[role == 'instructor'][0]{
+    'name': person->name,
+    'avatar_url': person->image.url,
+    'url': person->slug.current
```

Comment:
I noticed that this url is `/instructors/:instructor-slug` where it should be `/q/resources-by-:instructor-slug